controllers/chat/getSignedChannelList.js

```diff
@@ -15,8 +15,15 @@ const MAX_ITERATIONS = 10;
  * @param {number} offset
  * @returns
  */
-const __queryChannelBuilder = async (client, userId, limit, offset) => {
-  const channels = await client.queryChannels({members: {$in: [userId]}}, [{last_message_at: -1}], {
+const __queryChannelBuilder = async (client, userId, limit, offset, last_message_at = null) => {
+  let filter = {members: {$in: [userId]}};
+  if (last_message_at) {
+    last_message_at = new Date(last_message_at);
+    last_message_at = last_message_at.toISOString();
+    filter = {members: {$in: [userId]}, last_message_at: {$gte: last_message_at}};
+  }
+
+  const channels = await client.queryChannels(filter, [{last_message_at: -1}], {
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `last_message_at` variable is being overwritten in line 21 and 22. This could lead to unexpected behavior if the original value of `last_message_at` is needed later in the function. Consider using a new variable instead.
```diff
-    last_message_at = new Date(last_message_at);
-    last_message_at = last_message_at.toISOString();
+    const lastMessageDate = new Date(last_message_at);
+    const lastMessageISO = lastMessageDate.toISOString();
```
And then use `lastMessageISO` in the filter.
```diff
-    filter = {members: {$in: [userId]}, last_message_at: {$gte: last_message_at}};
+    filter = {members: {$in: [userId]}, last_message_at: {$gte: lastMessageISO}};
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->